src/inference/src/ie_core.cpp

```diff
@@ -886,6 +886,12 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
     ie::SoExecutableNetworkInternal ImportNetwork(std::istream& networkModel,
                                                   const std::string& deviceName,
                                                   const std::map<std::string, std::string>& config) override {
+        ie::CompiledBlobHeader header;
+        networkModel >> header;
+        if (header.getIeVersion() != ie::GetInferenceEngineVersion()->buildNumber) {
+            // Build number mismatch, the cached file can't be used
+            throw ie::NetworkNotRead("Version does not match");
+        }
```

Comment:
ImportNetwork is a public API, it's not guaranteed that it has the same compiled_blob created during the caching. In caching, compile_blob = header + orginal_compiled_blob.
With current change you are trying to find version in orginal_compiled_blob, while it should not be

Reply:
Okay, thanks for the explanation. Looks like this is a wrong way to fix the issue.
The original issue is exactly about compiled_blob representation. Network deserialization logic relies on the fact that it gets  compiled_blob without header (and if we use model caching, the header extracts from istream before network deserialization), however in case of direct import of the blob that was gotten with the help of model caching, the header remains in the istream and deserialization logic fails.
Probably the best idea is to handle the case with the header in the deserialization logic